src/Microsoft.Framework.DependencyInjection/Internal/ConcurrentDictionaryExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+namespace System.Collections.Concurrent
+{
+    internal static class ConcurrentDictionaryExtensions
+    {
+        // From https://github.com/dotnet/corefx/issues/394#issuecomment-69494764
+        // This lets us pass a state parameter allocation free GetOrAdd
+        internal static TValue GetOrAdd<TKey, TValue, TArg>(this ConcurrentDictionary<TKey, TValue> dictionary, TKey key, Func<TKey, TArg, TValue> valueFactory, TArg arg)
+        {
+            if (dictionary == null)
```

Comment:
If it's internal code, consider removing these exceptions and replacing it with `Debug.Assert` instead.